packages/dart/sshnoports/bundles/universal.sh

```diff
@@ -849,6 +849,42 @@ device() {
     device_install_type=$device_type
   fi
 
+  # install at_activate binary
+  "$extract_path"/sshnp/install.sh -b "$bin_path" -u "$user" at_activate
+
+  # install sshnpd binary and capture the installer output
+  install_output=$("$extract_path"/sshnp/install.sh -b "$bin_path" -u "$user" "$device_install_type" sshnpd)
+
+  if [ "$verbose" = true ]; then
+    echo "$install_output"
+  fi
+
+  # upgrade an existing installation if we find one
+  case "$device_install_type" in
+    launchd)
+      launchd_plist="$HOME/Library/LaunchAgents/com.atsign.sshnpd.plist"
+      if [ -f "$launchd_plist" ]; then
+        echo "launchd config already in place"
+        # TODO: restart service?
```

Comment:
@XavierChanth what would be the appropriate command here?

Reply:
From what I grok in the launchctl man page it's supposed to be this:

```
launchctl disable gui/$UID/com.atsign.sshnpd
launchctl enable gui/$UID/com.atsign.sshnpd
```

And I get a success code from both commands, but neither is actually stopping or starting the service...

Reply:
Ah, that doesn't work.. but this works:

`launchctl kickstart -k gui/$UID/com.atsign.sshnpd`

-k says to kill the running instance first.

Reply:
Confirmed by checking process ids for sshnpd